lib/scope.js

```diff
@@ -1759,8 +1759,13 @@ module.exports = {
           let name = await label.evaluate( elem => { return elem.getAttribute('for'); });
         }
 
-        if ( type === `radio` ) { await label.evaluate( elem => { return elem.click(); }); }
-        else { await scope.setCheckbox( scope, { label, answer }); }
+        if ( type === `radio` ) {
+          if (answer.toLowerCase() == "true") {
```

Comment:
After a bit of trial and error, I have a guess as to what happened here. In the non-random flow, if a radio button is going to be pressed, only that radio button will get into here. The `answer` is the DOM `value` of the radio button. A an `answer` for a 'no' radio button will probably be "False". If it's the set of choices, it'll be something like "radio_other_opt_3".

That is, by the time the handle is in here, this radio button should definitely be pressed, no matter what the answer is.

If we want to control the flow of random radio buttons, I think that needs to happen upstream where random answers are dealt with. If we don't want a radio button to be pressed, it shouldn't get sent in here.

Reply:
Hm, I get it. A weird place for a parameter to have no effect, but I don't know if there's a refactor that makes it cleaner.

From what I can tell, there's no existing function to get the elements of a radio button? i.e. which values it can be set to? I'll have to let this sit until I can figure that out.

Reply:
Discussion about this ~35 minutes in Deep dive recording